utils/util.go

```diff
@@ -0,0 +1,33 @@
+// Package util includes common, shared functions
+package util
+
+import (
+	"database/sql"
+	"math/rand"
+	"time"
+
+	"go.uber.org/zap"
+)
+
+// RetryDBCommitMaxSleep - max sleep time during DB commit retries, in seconds
+const RetryDBCommitMaxSleep = 3
+
+// RetryDBCommit commits a db transaction, using up to maxRetries retries if necessary
+func RetryDBCommit(tx *sql.Tx, maxRetries int, logger *zap.Logger) error {
+	var err error
+	for i := 0; i < maxRetries; i++ {
+		err = tx.Commit()
+		if err == nil {
+			return nil
+		}
+
+		logger.Sugar().Warn("Error while trying to commit a db transaction, retry #", i, "of ", maxRetries, ". Error is: ", err)
```

Comment:
This should be a more structure log entry instead of concatenated like this.

```suggestion
		logger.Warn("Unable to commit database transaction",
			zap.Int("attempts", i),
			zap.Int("max_attempts", maxRetries),
			zap.Error(err),
		)
```

if you wanted to stick with sugar you can use.

```suggestion
		logger.Sugar().Warnw("Unable to commit database transaction",
			"attempts", i,
			"max_attempts", maxRetries,
			"error", err,
		)
```

Reply:
Thanks, updated.